x-pack/solutions/observability/plugins/streams_app/public/components/entity_detail_view/index.tsx

```diff
@@ -120,7 +120,11 @@ export function EntityDetailViewWithoutParams({
                       </EuiBadge>
                     </>
                   ) : null}
-                  {definition && <LifecycleBadge lifecycle={definition.effective_lifecycle} />}
+                  {definition && (
+                    <LifecycleBadge
+                      lifecycle={definition.effective_lifecycle as IngestStreamLifecycle}
```

Comment:
Could we introduce an `EffectiveIngestStreamLivecycle` type and use it as the base for the type guards, then change `LifecycleBadge` to take that? This would get rid of this type cast here.

Reply:
Good call, fixed in https://github.com/elastic/kibana/pull/208500/commits/59fd26b7fbe465d4b4cd8f233ef860878ecf3d45